### DISCUSSION TITLE
Sub-menu in existing menu list

### DISCUSSION THREAD

**Discussion by hemantwasthere** [vidstack/player#1383] [Q&A]
I wanted to add a new sub-menu in the default existing menu list. 

- I am trying to make a video quality sub-menu like yt player has.

```typescript
 				<Menu.Root className="vds-menu">
                      <Menu.Button
                        className="vds-menu-button vds-button"
                        aria-label="Settings"
                      >
                        <SettingsIcon className="vds-rotate-icon vds-icon" />
                      </Menu.Button>
                      <Menu.Items
                        className="vds-menu-items"
                        placement="top"
                        offset={0}
                      >
                        <QualitySubmenu />
                      </Menu.Items>
               </Menu.Root>
``` 
- With this code, I can change the quality of the video with a separate setting button.
- But I want to add this sub-menu in a existing menu list. I looked into the docs but couldn't really find something useful to this.
- Any help would be appreciated 🙏

**Comment by mihar-22** [comment]
You would have to build out the rest of the menu for your own layout, otherwise, you can also slot it inside the Default Layout:

```tsx
<DefaultVideoLayout 
  slots={{ 
    settingsMenuItemsEnd: () => <QualitySubmenu /> 
  }} 
/>
```
